virttest/utils_test/__init__.py

```diff
@@ -242,11 +242,16 @@ def get_time(session, time_command, time_filter_re, time_format):
                 raise exceptions.TestError(msg)
             if re.search('PM', guest_time):
                 hour = re.findall('\d+ (\d+):', guest_time)[0]
-                hour = str(int(hour) + 12)
+                if int(hour) < 12:
+                    hour = str(int(hour) + 12)
                 guest_time = re.sub('\s\d+:', " %s:" % hour,
                                     guest_time)[:-3]
             else:
-                guest_time = guest_time[:-3]
+                hour = re.findall('\d+ (\d+):', guest_time)[0]
```

Comment:
`else` here include 24 hour format  and 12 hour format with "AM".  and this change only suitable for the second scenario.

Reply:
`hour = re.findall('\d+ (\d+):', guest_time)[0]`  we move it out of if-else, as it is used not only once.